platforms/documentation/docs/src/docs/userguide/reference/gradle_daemon.adoc

```diff
@@ -231,21 +245,57 @@ On the next execution of the build, the Gradle client will use this file to loca
+Gradle recognizes a small number of JVM vendor strings as special and equivalent.
+For example, "Adoptium" and "Temurin" are considered the same vendor.
+You can see the list of special vendors by running `gradle help --task updateDaemonJvm`.
 
-Gradle recognizes a small number of JVM vendor strings as special and equivalent. For example, "Adoptium" and "Temurin" are considered the same vendor. You can see the list of special vendors by running `gradle help --task updateDaemonJvm`.
+If the JVM vendor you specify is not treated as a special value, Gradle considers the value as an exact match.
+For example, to match the vendor "My Custom JVM", the vendor criterion must be "My Custom JVM".
 
-If the JVM vendor you specify is not treated as a special value, Gradle considers the value as an exact match. For example, to match the vendor "My Custom JVM", the vendor criterion must be "My Custom JVM".
+Configuring the vendor on the task itself uses the <<toolchains.adoc#sec:vendors,same syntax as the Java toolchain spec>>.
 
+[[daemon_jvm_discovery]]
 === Daemon JVM discovery
 
 To locate a compatible JVM installation, Gradle re-uses the mechanism provided by the <<toolchains.adoc#toolchains,Java Toolchains>> feature.
 This feature is used to locate a JVM installation that matches the criteria specified in the `gradle/gradle-daemon-jvm.properties` file.
 
-NOTE: The _daemon JVM discovery process_ does not support auto-provisioning of new JVM installations.
-This will be added in a future release.
+This process fully supports both <<toolchains.adoc#sec:auto_detection,auto-discovery>> and <<toolchains.adoc#sec:provisioning,auto-provisioning>>.
+
+[[daemon_jvm_provisioning]]
+==== Controlling auto-provisioning URLs
+
+The `updateDaemonJvm` task will, by default, attempt to generate download URLs for JDKs on different platforms (OS and architecture) that match the criteria.
+
+By default, Gradle will consider the following build platforms:
+
+* [`FREE_BSD`, `AARCH64`]
+* [`LINUX`, `AARCH64`]
+* [`LINUX`, `X86_64`]
+* [`MAC_OS`, `AARCH64`]
+* [`MAC_OS`, `X86_64`]
+* [`UNIX`, `AARCH64`]
+* [`UNIX`, `X86_64`]
+* [`WINDOWS`, `X86_64`]
+
+These can be configured through the `toolchainPlatforms` property on the `UpdateDaemonJvm` task.
+They do not have a CLI configuration option.
+
+TODO Add a snippet showing how to configure those and one on how to disable that feature
+
+Gradle will attempt to resolve JDKs URLs for these platforms by leveraging the <<toolchains.adoc#sub:download_repositories,configured toolchain download repositories>>.
+If no such repositories are configured, then the `updateDaemonJvm` task will fail.
+
+Instead of using toolchain download repositories, users may chose to configure the JDK URLs on the task itself, using the `toolchainDownloadUrls` property.
+This is a `Map<BuildPlatform, URI>` that can be configured as the following example shows:
+
+TODO add example of hardcoding URLs in task configuration
```

Comment:
See test [here](https://github.com/gradle/gradle/pull/32253/files#diff-461468b1cd1e6067768f462b00162ddecca3097cda2765f241c642cd2fd80f27R396) for an example